src/solve.c

```diff
@@ -130,3 +132,59 @@ void run(void)
 			return;
 	}
 }
+
+/**
+ * @brief Function to save the maze sequence on EEPROM.
+ */
+void save_maze(void)
+{
+	uint32_t save_status = 0;
+
+	save_status = eeprom_flash_page(FLASH_EEPROM_ADDRESS_MAZE,
+					(uint8_t *)run_sequence, MAZE_AREA);
+
+	if (save_status != RESULT_OK)
+		LOG_ERROR("EEPROM save error %" PRIu32, save_status);
+}
+
+/**
+ * @brief Function to load the maze sequence from EEPROM to static on RAM.
+ */
+void load_maze(void)
+{
+	eeprom_read_data(FLASH_EEPROM_ADDRESS_MAZE, MAZE_AREA,
+			 (uint8_t *)run_sequence);
+}
+
+/**
+ * @brief Function to reset the maze sequence on EEPROM.
+ */
+void reset_maze(void)
+{
+	uint32_t erase_status = 0;
+
+	erase_status = eeprom_erase_page(FLASH_EEPROM_ADDRESS_MAZE);
+	if (erase_status != RESULT_OK)
+		LOG_ERROR("EEPROM reset error %" PRIu32, erase_status);
+}
+
+/**
+ * @brief Function to check if the maze sequence is saved on EEPROM.
+ *
+ *@return bool
+ */
+bool maze_is_saved(void)
+{
+	uint8_t maze_sample[EEPROM_NUM_BYTES_ERASED_CHECKED];
+	bool is_saved = false;
+
+	eeprom_read_data(FLASH_EEPROM_ADDRESS_MAZE,
+			 EEPROM_NUM_BYTES_ERASED_CHECKED, maze_sample);
+
+	for (uint8_t iter = 0; iter < EEPROM_NUM_BYTES_ERASED_CHECKED; iter++) {
+		if (maze_sample[iter] != EEPROM_BYTE_ERASED_VALUE)
+			is_saved = true;
+	}
+
+	return is_saved;
```

Comment:
You do not even need `is_saved`. You could:

```c
	uint8_t maze_sample[EEPROM_NUM_BYTES_ERASED_CHECKED];

 	eeprom_read_data(FLASH_EEPROM_ADDRESS_MAZE,
			 EEPROM_NUM_BYTES_ERASED_CHECKED, maze_sample);
 	for (uint8_t iter = 0; iter < EEPROM_NUM_BYTES_ERASED_CHECKED; iter++) {
		if (maze_sample[iter] != EEPROM_BYTE_ERASED_VALUE)
			return true;
	}
 	return false;
```

I still think there is no benefit in reading multiple bytes. :joy: